eloquentjs/4_4_deep_comparison.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Performs deep comparison of two types and determines equality
+ * @param {*} itemA 
+ * @param {*} itemB 
+ * @returns 
+ */
+function deepEqual(itemA, itemB) {
+    if(itemA === null && itemB === null) {
```

Comment:
Perhaps this should just be if itemA === itemB return true? Referential equality is certainly deep equality - it's the same molecules in the computer!